processor/logdedupprocessor/counter.go

```diff
@@ -144,21 +162,32 @@ func (a *logCounter) Increment() {
 	a.count++
 }
 
-// getLogKey creates a unique hash for the log record to use as a map key
-/* #nosec G104 -- According to Hash interface write can never return an error */
-func getLogKey(logRecord plog.LogRecord) [8]byte {
-	hasher := fnv.New64()
-	attrHash := pdatautil.MapHash(logRecord.Attributes())
+// getResourceKey creates a unique hash for the resource to use as a map key
+func getResourceKey(resource pcommon.Resource) uint64 {
+	hasher := xxhash.New()
+	attrsHash := pdatautil.MapHash(resource.Attributes())
+	_, _ = hasher.Write(attrsHash[:])
+	return hasher.Sum64()
+}
 
-	hasher.Write(attrHash[:])
+// getScopeKey creates a unique hash for the scope to use as a map key
+func getScopeKey(scope pcommon.InstrumentationScope) uint64 {
+	hasher := xxhash.New()
+	attrsHash := pdatautil.MapHash(scope.Attributes())
+	_, _ = hasher.Write(attrsHash[:])
+	_, _ = hasher.Write([]byte(scope.Name()))
+	_, _ = hasher.Write([]byte(scope.Version()))
+	return hasher.Sum64()
+}
+
+// getLogKey creates a unique hash for the log record to use as a map key
+func getLogKey(logRecord plog.LogRecord) uint64 {
```

Comment:
I see why this one isn't standard, since we're only accounting for some fields. Maybe an options pattern accounts for it in the future, but I'm find with leaving it alone for now.